samples/nrf5340/multicore/README.rst

```diff
@@ -0,0 +1,65 @@
+.. _nrf5340_multicore:
+
+nRF5340: Multi core application
+###############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can use this sample to see how to build an application with the network core build as child image from soruce.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf5340dk_nrf5340_cpuapp
+
+Overview
+********
+
+The sample application shows how to build multi core application, with use of ``add_child_image`` cmake macro.
+Note the application source dir structure and additional files:
+
+   * ``aci/CMakeLists.txt`` - adding child image.
+   * ``zephyr/module.yml`` - information about location of ``CMakeLists.txt`` file that invokes ``add_child_image``.
+     Required for build system.
+   * Custom ``Kconfig`` file that allows to add child image only for application core.
+     Additionally it enables options required by multi image build.
+
+Both cores uses the same ``main.c``, which prints name of the board its run on.
+
+For more information about multi image builds, see :ref:`ug_multi_image`.
+
+Building and running
```

Comment:
In the multi-image user guide, we describe changes in the build process related to using multi-image. How does using multi-image in this sample affect the building process?

Reply:
That's a bit complicated, i will try to be brief.

Once you call `find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})` in Applications `CMakeLists.txt`, NCS repository is scanned for related dependencies, child_images in particular. Child images are selected (and build) depending on selected options in root image (for our case its application image). Apart from child image mechanism, NCS has many others, e.g. Partition manager.

In this build system one thing was not considered - if the user wants to use `add_child_image` by himself (from applications (`CMakeLists.txt`). Because of how build system is build without those "magic files" i've added (`aci/CMakeLists.txt` and `zephyr/module.yml`), the build invokes some cmake files in wrong order (partition manager). What results in build error.

What this sample (and "magic files") does, it tricks the build system so that it call `partition_manager.cmake` alongside other files (`multi_image.cmake`) in correct order. That allows to build the application correctly.

I hope that clears things a bit.

Reply:
Thanks! Let's add this explanation to line 36, after the new line 35.

```
When building any application for the nRF5340 SoC, the build system adds the child images based on the options selected for the parent image.
This sample overrides this process by specifying its dedicated sources for the child images.
To do this, it points the build system to the correct child image location in the :file:`aci/CMakeLists.txt` file by using :file:`zephyr/module.yml`.
```